okhttp/src/main/java/okhttp3/Handshake.kt

```diff
@@ -24,50 +24,96 @@ import javax.net.ssl.SSLPeerUnverifiedException
 import javax.net.ssl.SSLSession
 
 /**
- * A record of a TLS handshake. For HTTPS clients, the client is *local* and the remote server
- * is its *peer*.
+ * A record of a TLS handshake. For HTTPS clients, the client is *local* and the remote server is
+ * its *peer*.
  *
- * This value object describes a completed handshake. Use [ConnectionSpec] to set policy
- * for new handshakes.
+ * This value object describes a completed handshake. Use [ConnectionSpec] to set policy for new
+ * handshakes.
  */
-data class Handshake private constructor(
-  private val tlsVersion: TlsVersion,
-  private val cipherSuite: CipherSuite,
-  private val peerCertificates: List<Certificate>,
-  private val localCertificates: List<Certificate>
-) {
-
+class Handshake private constructor(
```

Comment:
Could still have ben a `data class` no?

Reply:
I think so. Biggest drawback of data class is that it’ll accept mutable lists, particularly from Java.

Reply:
(I’m gonna punt on data class or not in public APIs for now.)